proto/redpanda/api/dataplane/v1/knowledge_base.proto

```diff
@@ -0,0 +1,75 @@
+syntax = "proto3";
+
+package redpanda.api.dataplane.v1;
+
+import "buf/validate/validate.proto";
+import "google/api/field_behavior.proto";
+
+// KnowledgeBase defines the configuration and metadata for a retrieval-augmented generation (RAG) Knowledge Base.
+message KnowledgeBase {
+  // KnowledgeBase ID.
+  string id = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.pattern = "^[A-Za-z0-9-_/]+$"
+  ];
```

Comment:
We should autogenerate the id and not have it user-specified I think. 

Also this id will likely go into a Kafka topic name (e.g. `__redpanda_cloud__rag_<kb-id>_source_confluence-with-id`) so we need to limit the character set. Slashes wouldn't be allowed.

Reply:
oh yeah, absolutely. my mistake. should be xid

Reply:
this is the resource. so it's required, because it's never an input